test/test.js

```diff
@@ -458,6 +462,43 @@ function checkRefTestResults(browser, id, results) {
   });
 }
 
+function browserTestReportHandler(req, res) {
+  var parsedUrl = url.parse(req.url, true);
+  var pathname = parsedUrl.pathname;
+  if (pathname === '/browserTestReports' ||
+      pathname === '/browserWorkerTestReports') {
+
+    var writableStream = fs.createWriteStream(
+                           '../coverage/coverageinfo.json');
+    var body = '';
+
+    req.on('data', function(chunk) {
+      body += chunk;
+      writableStream.write(chunk);
+    });
+    req.on('end', function() {
+      writableStream.end();
+    });
+
+    var map = libCoverage.createCoverageMap(JSON.parse(body));
```

Comment:
This won't work: `body` will be an empty string at this point. You should only start parsing the response in the `end` event of `res`.

It is probably slow and difficult to test the whole system at once, so I suggest that you test the server in isolation while developing it. E.g. by copying your `XMLHttpRequest` code to a single HTML file, and use that HTML file to initiate requests. Or by using `curl`.

If you don't really understand what your code is doing, consider stepping through it with a debugger to see what your code is doing. Here is a guide: https://medium.com/@paul_irish/debugging-node-js-nightlies-with-chrome-devtools-7c4a1b95ae27

Reply:
> It is probably slow and difficult to test the whole system at once, so I suggest that you test the server in isolation while developing it. E.g. by copying your XMLHttpRequest code to a single HTML file, and use that HTML file to initiate requests. Or by using curl.

I'll keep this in mind when I integrate the JSON to HTML feature, till that time, do we need any more changes ?